vm/src/hint_processor/builtin_hint_processor/bootloader_hints.rs

```diff
@@ -0,0 +1,191 @@
+        ap_tracking,
+    )?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::hint_processor::builtin_hint_processor::hint_utils::get_maybe_relocatable_from_var_name;
+    use crate::hint_processor::hint_processor_definition::HintReference;
+    use crate::types::exec_scope::ExecutionScopes;
+    use crate::types::relocatable::MaybeRelocatable;
+    use crate::utils::test_utils::*;
+    use crate::vm::runners::builtin_runner::BuiltinRunner;
+    use crate::vm::vm_core::VirtualMachine;
+
+    use super::*;
+
+    #[test]
+    fn test_prepare_simple_bootloader_output_segment() {
+        let mut vm = vm!();
+        vm.segments.add();
+        vm.run_context.fp = 1;
+
+        let mut output_builtin = OutputBuiltinRunner::new(true);
+        output_builtin.initialize_segments(&mut vm.segments);
+        vm.builtin_runners
+            .push(BuiltinRunner::Output(output_builtin.clone()));
+
+        let mut exec_scopes = ExecutionScopes::new();
+        let ids_data = ids_data!["simple_bootloader_output_start"];
+        let ap_tracking = ApTracking::new();
+
+        let bootloader_input = BootloaderInput {
+            bootloader_config: BootloaderConfig {
+                simple_bootloader_program_hash: ProgramHash(1234),
+                supported_cairo_verifier_program_hashes: vec![ProgramHash(5678), ProgramHash(8765)],
+            },
+            packed_outputs: vec![],
+        };
+
+        exec_scopes.insert_value(vars::BOOTLOADER_INPUT, bootloader_input);
+        prepare_simple_bootloader_output_segment(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking).expect("Hint failed unexpectedly");
+
+        let current_output_builtin = vm
+            .get_output_builtin()
+            .expect("The VM should have an output builtin").clone();
+        let stored_output_builtin: OutputBuiltinRunner = exec_scopes
+            .get(vars::OUTPUT_BUILTIN_STATE)
+            .expect("The output builtin is not stored in the execution scope as expected");
+
+        // Check the content of the stored output builtin
+        assert_ne!(current_output_builtin.base(), stored_output_builtin.base());
+        assert_eq!(stored_output_builtin.base(), output_builtin.base());
+        assert_eq!(stored_output_builtin.stop_ptr, output_builtin.stop_ptr);
+        assert_eq!(stored_output_builtin.included, output_builtin.included);
+
+        let simple_bootloader_output_start = get_maybe_relocatable_from_var_name(
+            "simple_bootloader_output_start",
+            &vm,
+            &ids_data,
+            &ap_tracking,
+        )
+        .expect("Simple bootloader output start not accessible from program IDs");
+        match simple_bootloader_output_start {
+            MaybeRelocatable::RelocatableValue(relocatable) => {
+                assert_eq!(relocatable.segment_index, current_output_builtin.base() as isize);
+            }
+            other => panic!("Expected a relocatable value but got {:?}", other)
+        }
```

Comment:
`matches!()` might express this better, but maybe not...

https://doc.rust-lang.org/std/macro.matches.html